src/Components/Components/src/Dispatcher.cs

```diff
@@ -23,6 +23,20 @@ public abstract class Dispatcher
         /// </summary>
         internal event UnhandledExceptionEventHandler UnhandledException;
 
+        /// <summary>
+        /// Validates that the currently executing code is running inside the dispatcher.
+        /// </summary>
+        public void AssertAccess()
+        {
+            if (!CheckAccess())
+            {
+                throw new InvalidOperationException(
```

Comment:
I'm not a super fan of this API. Telling the user to do this isn't helpful all the time - for example, user code is not involved in the `CircuitHost` where this is used, so throwing an exception stating they should do something when it's our bug doesn't seem right.

Could we write something specific in the host instead?

Reply:
It won't happen though. 

We could do *extra* work to have a worse error message for a case that should never happen. Or we can call the API that does what we want, with the expectation that it won't happen.